xwiki-rendering-macros/xwiki-rendering-macro-ctsreport/src/main/java/org/xwiki/rendering/internal/macro/ctsreport/Result.java

```diff
@@ -59,15 +59,12 @@ public String toString()
     @Override
     public boolean equals(Object object)
     {
-        if (object == null) {
+        if (object == null || object.getClass() != getClass()) {
```

Comment:
Since Result is not really designed to be extended, a slight improvement would be

    if (!(object instanceof Result))

(it cannot be an instance of Result if it's null)